### ISSUE TITLE
GDScript: Add a new annotation to make direct calls into native libraries without writing an extension

### ISSUE THREAD

**Issue by MarioLiebisch** [godotengine/godot-proposals#8348]
### Describe the project you are working on

Writing a demo application for our commercial native library. I wrote a custom GDExtension to work around the limitation, but would have been awesome to just skip that step.

### Describe the problem or limitation you are having in your project

Currently there's no quick and simple way to call native libraries from GDScript code. While C# has the possibility to use .net's own Interop library, there's no GDScript equivalent.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

With this feature implemented, it would be possible to quickly call native functions, e.g. from system libraries, without using C# or specifically writing a custom GDExtension for this.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I imagine this feature as a new annotation for function definitions that would make the actual function into a fallback only, if the target library couldn't be found or the feature simply isn't supported on a platform. Otherwise the native function would be called instead of the function body.

```swift
@native("User32", "MessageBox")
func MessageBox(hwnd: int, text: String, caption: String, style: int = 0) -> int:
  var dialog = AcceptDialog.new()
  dialog.dialog_text = text
  dialog.window_title = title
  dialog.connect('modal_closed', dialog, 'queue_free')
  add_child(dialog)
  dialog.popup_centered()
  return 0
```

Now if you'd call this function on Windows:
* Try to load "User32.dll".
* Try to retrieve the function pointer for "MessageBox".
* If this fails, retry with "MessageBoxW".
* If this fails, retry with "MessageBoxA".
* If all fail, run the regular GDScript function body.
* If loading is successful, disregard the function body and run the native function instead. Function parameter types and return value are deducted from the GDScript function's signature. (This might need additional thoughts on more complex or ambiguous types).

If you call it on any other platform (or loading "User32" fails for whatever reason), you'd end up with a regular GDScript function running its body (i.e. the annotation does nothing in this case).

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I don't see any simple way around this directly. One would always have to fall back to C# or a GDExtension.

### Is there a reason why this should be core and not an add-on in the asset library?

As far as I'm aware, it's not possible to extend/modify GDScript itself from an extension or addon, if one wants to use an annotation for this. It would certainly be possible to use a GDExtension adding some extra object type/class to handle this, obviously.

**Comment by AThousandShips** [comment]
Related?
* https://github.com/godotengine/godot/pull/85741

**Comment by MarioLiebisch** [comment]
> Related?
Yes, this covers pretty much the same idea. Might have a look later. (On my own research so far I also arrived at libffi, since other alternatives are outdated or incompatible license wise, but simply didn'T find the time to look deeper.)
